canfund-rs/src/manager/record.rs

```diff
@@ -127,6 +145,40 @@ impl CyclesBalance {
     }
 }
 
+/// The funding failure record.
+#[derive(Clone, Debug, Default, PartialEq, Eq)]
+pub struct FundingFailure {
+    /// The code of the reason for the funding failure.
+    pub error_code: FundingErrorCode,
+    /// The timestamp of the failure.
+    pub timestamp: u64,
+}
+
+#[derive(Clone, Debug, Default, PartialEq, Eq)]
+pub enum FundingErrorCode {
+    #[default]
+    InsufficientCycles, // Funding canister has insufficient cycles
+    DepositFailed,      // The deposit of cycles failed
+    MintingFailed,      // Minting cycles from ICP failed
```

Comment:
Should this be called ObtainFailed instead? Since currently this can be triggered also by cycles ledger withdrawal not just minting from ICP. Maybe add a reason payload for details.